### PR TITLE
Features/ft 835 merge hd export

### PR DESCRIPTION
**Pull request by roberthorlings**
Merge of rc2_dev export code into master branch, to enable exporting using the core-api

### CODE REVIEW

**Comment by cataphract** [comment]
This has direct references to core-db (packages `org.transmartproject.db.*`). core-db is a runtime dependency, even if Grails doesn't enforce the compile/runtime separation. Refactor these out, either by
1. Using core-api exposed functionality. Core-api classes are in `org.transmartproject.core.*`.
2. If functionality is not available there (or it's too inefficient), you may need to add it to core-api.
3. As a last resort, move the functionality to transmartApp. This should be avoided; we want to route as much functionality through core-api as possibly so that we can for instance enforce security constraints centrally.

**Comment by cataphract** [comment]
Reorganize imports. A lot of them are clearly unnecessary just by looking at the diff (for instance there are imports to `java.util.*` classes).

**Comment by cataphract** [comment]
`DataCountService` still has references to core-db.

**Comment by cataphract** [comment]
Can you merge with master? Seems that could be the reason the travis build is failing.

**Comment by roberthorlings** [comment]
I will. Another reason that travis is failng is that it used a branch from transmart-core-api which contained a method that was not implemented. but I'll merge with master

**Comment by cataphract** [comment]
I still see an import of a core-db class and java.util classes.

File: grails-app/services/com/recomdata/asynchronous/GenericJobService.groovy [hunk 1]
```diff
@@ -51,16 +43,16 @@ import org.rosuda.REngine.Rserve.*;
...
-	
-	String tempFolderDirectory
+    def ctx = Holders.grailsApplication.mainContext
+    def springSecurityService = ctx.springSecurityService
+    def jobResultsService = ctx.jobResultsService
+    def i2b2HelperService = ctx.i2b2HelperService
+    def i2b2ExportHelperService = ctx.i2b2ExportHelperService
+    def snpDataService = ctx.snpDataService
+    def dataExportService = ctx.dataExportService
+    def asyncJobService = ctx.asyncJobService
```
Comment (cataphract) [inline grails-app/services/com/recomdata/asynchronous/GenericJobService.groovy:53]: What happened here? Why is injection not used anymore?

File: grails-app/services/com/recomdata/asynchronous/GenericJobService.groovy [hunk 2]
```diff
@@ -235,45 +226,36 @@ class GenericJobService implements Job {
...
+                                }
 
-                        if (StringUtils.isEmpty(remoteFilePath)) {
-                            break
+                                //Delete the ZIP file
+                                outputFile.delete();
+                            }
                         }
+                    } catch (Exception e) {
+                        println("Failed to FTP PUT the ZIP file");
```
Comment (cataphract) [inline grails-app/services/com/recomdata/asynchronous/GenericJobService.groovy]: println

File: grails-app/services/com/recomdata/transmart/data/export/DataExportService.groovy [hunk 3]
```diff
@@ -100,14 +110,32 @@ class DataExportService {
...
 
 						def List gplIds = subsetSelectedPlatformsByFiles?.get(subset)?.get(selectedFile)
-						def retVal
-						switch (selectedFile)
-						{
+                        def retVal = null
+                        switch (selectedFile) {
 							case "STUDY":
 								retVal = metadataService.getData(studyDir, "experimentalDesign.txt", jobDataMap.get("jobName"), studyList);
+                                log.info("retrieved study data")
```
Comment (cataphract) [inline grails-app/services/com/recomdata/transmart/data/export/DataExportService.groovy]: This is at most a debug statement, trace or no statement at all would be even better..

**Pull request merged**